src/RestApi/StoreApi/Utilities/MoneyValue.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Handles monetary values using integers by using the smallest unit of a currency.
+ *
+ * This could be replaced with https://github.com/moneyphp/money if we want to
+ * support currencies in the future. However at this stage we only need to use
+ * the store decimal setting (which is independent of currency) to format
+ * values for the API.
+ *
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\RestApi\StoreApi\Utilities;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * MoneyValue class.
+ */
+class MoneyValue {
+
+	/**
+	 * Monetary amount. Integer stored as string.
+	 *
+	 * @var string
+	 */
+	protected $amount;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string|integer $amount Set money amount.
+	 */
+	public function __construct( $amount = '0' ) {
+		$this->amount = (string) $amount;
+	}
+
+	/**
+	 * Converts to string if this class is used as a string.
+	 */
+	public function __toString() {
+		return $this->amount;
+	}
+
+	/**
+	 * Return the price amount.
+	 *
+	 * @return string|int
+	 */
+	public function get_amount() {
+		return $this->amount > PHP_INT_MAX ? $this->amount : (int) $this->amount;
+	}
+
+	/**
+	 * Set monetary amount in subunit format from decimal format.
+	 *
+	 * @param string|float $amount Monetary amount with decimals.
+	 * @param int          $decimals Number if decimals the amount is formatted with.
+	 * @param string       $rounding_mode Valid rounding mode: PHP_ROUND_HALF_DOWN, PHP_ROUND_HALF_UP, PHP_ROUND_HALF_EVEN, PHP_ROUND_HALF_ODD.
+	 * @return string      The new amount.
+	 */
+	public function from_decimal( $amount, $decimals = 2, $rounding_mode = 'PHP_ROUND_HALF_UP' ) {
+		$this->amount = (string) intval(
```

Comment:
Value objects should never be mutated.  I would expect a implementation more like:

```php
// where $currency is an instance of `CurrencyValue` object I mentioned in an earlier comment
( new MoneyValue( 123, $currency ) )->with_precision();
```

Or you could have a helper that returns a MoneyValue instance from a string so:

```php
( MoneyValue::from_string( '30.20', $currency ) )->with_precision();
```

If you wanted to convert the Money Value to a different currency (which isn't needed yet) you might have a method that would do something like this:

```php
$moneyValue->to_currency( $new_currency_value_instance )->with_precision()
```

Thoughts?

Reply:
As discussed, replacing this with simpler method specifically for API responses.